R/redist_smc_tidy.R

```diff
@@ -0,0 +1,226 @@
+#'
+#' This function draws nearly-independent samples from a specific target measure,
+#' controlled by the \code{popcons}, \code{compactness}, \code{constraints}, and
+#' \code{constraint_fn} parameters.
+#'
+#' Key to ensuring good performance is monitoring the efficiency of the resampling
+#' process at each SMC stage.  Unless \code{silent=F}, this function will print
+#' out the effective sample size of each resampling step to allow the user to
+#' monitor the efficiency.  If \code{verbose=T} the function will also print
+#' out information on the \eqn{k_i} values automatically chosen and the
+#' acceptance rate (based on the population constraint) at each step.
+#'
+#' Higher values of \code{compactness} sample more compact districts;
+#' setting this parameter to 1 is computationally efficient and generates nicely
+#' compact districts.  Values of other than 1 may lead to highly variable
+#' importance sampling weights.  By default these weights are truncated using
+#' \code{\link{redist_quantile_trunc}} to stabilize the resulting estimates, but
+#' if truncation is used, a specific truncation function should probably be
+#' chosen by the user.
+#'
+#' The \code{constraints} parameter allows the user to apply several common
+#' redistricting constraints without implementing them by hand. This parameter
+#' is a list, which may contain any of the following named entries:
+#' * \code{status_quo}: a list with two entries:
+#'   * \code{strength}, a number controlling the tendency of the generated districts
+#'   to respect the status quo, with higher values preferring more similar
+#'   districts.
+#'   * \code{current}, a vector containing district assignments for
+#'   the current map.
+#' * \code{vra}: a list with five entries, which may be set up using
+#'   \code{\link{redist.constraint.helper}}:
+#'   * \code{strength}, a number controlling the strength of the Voting Rights Act
+#'   (VRA) constraint, with higher values prioritizing majority-minority districts
+#'   over other considerations.
+#'   * \code{tgt_vra_min}, the target percentage of minority voters in minority
+#'   opportunity districts. Defaults to 0.55.
+#'   * \code{tgt_vra_other} The target percentage of minority voters in other
+#'   districts. Defaults to 0.25, but should be set to reflect the total minority
+#'   population in the state.
+#'   * \code{pow_vra}, which controls the allowed deviation from the target
+#'   minority percentage; higher values are more tolerant. Defaults to 1.5
+#'   * \code{min_pop}, A vector containing the minority population of each
+#'   geographic unit.
+#' * \code{incumbency}: a list with two entries:
+#'   * \code{strength}, a number controlling the tendency of the generated districts
+#'   to avoid pairing up incumbents.
+#'   * \code{incumbents}, a vector of precinct indices, one for each incumbent's
+#'   home address.
+#'
+#'
+#' @param map A \code{\link{redist_map}} object.
+#' @param n_sims The number of samples to draw.
```

Comment:
This should be renamed as `nsims`

Reply:
You're killing me.  Fixed in next commit